craft_providers/multipass/multipass.py

```diff
@@ -386,3 +388,105 @@ def umount(self, *, mount: str) -> None:
                 brief=f"Failed to unmount {mount!r}.",
                 details=errors.details_from_called_process_error(error),
             ) from error
+
+    def wait_until_ready(
+        self, *, retry_wait: float = 0.25, timeout: Optional[float] = None
+    ) -> Tuple[str, Optional[str]]:
+        """Wait until Multipass is ready (upon install/startup).
+
+        :param retry_wait: Time to sleep between retries.
+        :param timeout: Optional time.time() stamp to timeout after.
+
+        :returns: Tuple of parsed versions (multipass, multipassd).  multipassd
+            may be None if Multipass is not yet ready.
+        """
+        if timeout is not None:
+            deadline: Optional[float] = time.time() + timeout
+        else:
+            deadline = None
+
+        while True:
+            multipass_version, multipassd_version = self.version()
```

Comment:
why don't you try and catch this exception here? I guess we can wait and see what happens

Reply:
Which exception is that?  version() may raise MultipassError() which can't be handled (unexpected error cases).